plugins/inputs/dns/dns.go

```diff
@@ -0,0 +1,143 @@
+package dns
+
+import (
+	"errors"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/miekg/dns"
+	"net"
+	"strconv"
+	"time"
+)
+
+type Dns struct {
+	// Domains or subdomains to query
+	Domains []string
+
+	// Server to query
+	Servers []string
+
+	// Record type
+	RecordType string
+
+	// DNS server port number
+	Port int
+
+	// Dns query timeout in seconds. 0 means no timeout
+	Timeout int
+}
+
+var sampleConfig = `
+  ### Domains or subdomains to query
+  domains = ["mjasion.pl"] # required
+
+  ### servers to query
+  servers = ["8.8.8.8"] # required
+
+  ### Query record type. Posible values: A, CNAME, MX, TXT, NS. Default is "A"
+  recordType = "A" # optional
+
+  ### Dns server port. 53 is default
+  port = 53 # optional
+
+  ### Query timeout in seconds. Default is 2 seconds
+  timeout = 2 # optional
+`
+
+func (d *Dns) SampleConfig() string {
+	return sampleConfig
+}
+
+func (d *Dns) Description() string {
+	return "Query given DNS server and gives statistics"
+}
+func (d *Dns) Gather(acc telegraf.Accumulator) error {
+	d.setDefaultValues()
+	for _, domain := range d.Domains {
+		for _, server := range d.Servers {
+			dnsQueryTime, err := d.getDnsQueryTime(domain, server)
+			if err != nil {
+				return err
+			}
+			tags := map[string]string{
+				"server":     server,
+				"domain":     domain,
+				"recordType": d.RecordType,
+			}
+
+			acc.Add("dns", dnsQueryTime, tags)
```

Comment:
this is just in case we would like to add more metrics to this plugin in the future